productivity-suite/app/legacy-app/src/components/pages/News.tsx

```diff
@@ -11,9 +16,54 @@ export function News() {
     };
   });
 
+  useEffect(() => {
+    if (currentPageRef.current === undefined) {
+      const pageDetails = getBus().latestValue<{ page: number }>(
+        "change-news-page"
+      );
+      if (
+        pageDetails?.page !== undefined &&
+        currentPageRef.current !== pageDetails.page
+      ) {
+        currentPageRef.current = pageDetails.page;
+        setSearchParams({
+          page: `${pageDetails.page}`,
+        });
+      }
+    }
+
+    // reset the page 1 one when soft navigating to some other section of the app
+    return () => getBus().dispatch("news-page", { page: 1 });
+  }, []);
+
+  useEffect(() => {
+    const page = getPageNumberFromSearchParams(searchParams);
+    getBus().dispatch("news-page", { page });
+  }, [searchParams]);
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    return getBus().listen<{ page: number }>("change-news-page", ({ page }) => {
+      setSearchParams({
+        page: `${page}`,
+      });
+    });
+  }, [ref.current]);
+
   return (
     <div>
       <piercing-fragment-outlet fragment-id="news" />
     </div>
   );
 }
+
+function getPageNumberFromSearchParams(searchParams: URLSearchParams) {
+  const searchParamPage = searchParams.get("page");
+  const searchParamPageNum = searchParamPage ? +searchParamPage : 1;
+  const newsPageNum =
+    isFinite(searchParamPageNum) && searchParamPageNum > 0
+      ? searchParamPageNum
+      : 1;
+  return newsPageNum;
```

Comment:
the problem is that search parameters can be defined directly by the user, so that's why I wanted to put extra checks in place

for example in your code, I think that if we had `?page=hello`, we would end up with `NaN` as the `newsPageNum` instead of gracefully falling back to `1`

Reply:
I tried putting invalid search params and it seems to fallback to the first page, no problem...

Reply:
yeah you're right 😅 

it does happen to work properly but I think it's only because down the line in the fragment we check for the page's validity

you can see that we do end up with `NaN` in the message bus:
![Screenshot 2022-11-16 at 13 54 05](https://user-images.githubusercontent.com/61631103/202199165-3d70d4d2-d03b-44e0-966b-31e814d2db3d.png)

<details>
<summary>the 2 `{"page": null}` logs</summary>
the 2 `{"page": null}` logs there are from the news fragment, that's because `JSON.stringify` treats `NaN` as `null`:
![Screenshot 2022-11-16 at 13 56 02](https://user-images.githubusercontent.com/61631103/202199583-15e82279-61e0-4bd9-9472-b0b8ffcf5b70.png)
</details>

if we remove the check in the fragment things do start to break:
![Screenshot 2022-11-16 at 13 58 53](https://user-images.githubusercontent.com/61631103/202200316-9405450a-abba-4567-98e1-ff448a1790ba.png)

So it is not a big deal (one one ever will notice this anyways I think 😅), if you find the shorter code better I am fine with it, but I sort of feel like the code is not completely correct and we make up for it somewhere else

Reply:
Even the previous code is not immune to this. For example what about ?page=1.5?

Reply:
yeah I hadn't considered that 🤦 

I can amend the code to account for that as well (checking that the value is an integer)

But I get the feeling you do prefer the shorter version?
Personally I prefer the longer more robust one (although we are not writing a production application... 😅), but I am definitely not too set on it, I am totally fine with the shorter version (especially since it makes for simpler code in the demo, and we are also already covered as we mentioned)

Reply:
OK here is my proposed version:

```ts
function x(v) {
  const n = parseInt(v);
  return n > 0 ? n : 1;
}
```

This gives

```
> x()
1
> x("")
1
> x("foo")
1
> x("-1")
1
> x("-2")
1
> x("2.5")
2
> x("0")
1
```

Reply:
sounds good 😃👍